framework/src/modules/dpos_v2/events/delegate_voted.ts

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright © 2022 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ */
+
+import { BaseEvent, EventQueuer } from '../../base_event';
+import { ADDRESS_LENGTH } from '../constants';
+
+export const enum DelegateVotedEventResult {
+	SUCCESSFUL = 0,
+	FAIL_INSUFFICIENT_BALANCE = 1,
+	FAIL_RECIPIENT_NOT_INITIALIZED = 2,
+}
+
+export interface DelegateVotedEventData {
+	senderAddress: Buffer;
+	delegateAddress: Buffer;
+	amount: bigint;
+	result: DelegateVotedEventResult;
+}
+
+export const delegateVotedDataSchema = {
+	$id: '/pos/events/delegateVotedData',
+	type: 'object',
+	required: ['senderAddress', 'delegateAddress', 'amount', 'result'],
+	properties: {
+		senderAddress: {
+			dataType: 'bytes',
+			fieldNumber: 1,
+			minLength: ADDRESS_LENGTH,
+			maxLength: ADDRESS_LENGTH,
+		},
+		delegateAddress: {
+			dataType: 'bytes',
+			fieldNumber: 2,
+			minLength: ADDRESS_LENGTH,
+			maxLength: ADDRESS_LENGTH,
+		},
+		amount: {
+			datatype: 'uint64',
+			fieldNumber: 3,
+		},
+		result: {
+			dataType: 'uint32',
+			fieldNumber: 4,
+		},
+	},
+};
+
+export class DelegateVotedEvent extends BaseEvent<DelegateVotedEventData> {
```

Comment:
For the event which has `result` which can be used as persistent and non-persistent, it will be better to differentiate `log` and `error` similar to the token events